src/test/datascience/data-viewing/dataViewerDependencyService.unit.test.ts

```diff
@@ -4,118 +4,99 @@
+import * as helpers from '../../../kernels/helpers';
+import * as sinon from 'sinon';
 
 suite('DataScience - DataViewerDependencyService', () => {
     let dependencyService: DataViewerDependencyService;
     let appShell: IApplicationShell;
-    let pythonExecFactory: IPythonExecutionFactory;
-    let installer: IInstaller;
-    let interpreter: PythonEnvironment;
-    let interpreterService: IInterpreterService;
-    let pythonExecService: IPythonExecutionService;
+    let kernel: IKernel;
+
     setup(async () => {
-        interpreter = {
-            displayName: '',
-            uri: Uri.file(path.join('users', 'python', 'bin', 'python.exe')),
-            sysPrefix: '',
-            sysVersion: '',
-            version: new SemVer('3.3.3')
-        };
-        pythonExecService = mock<IPythonExecutionService>();
-        installer = mock(ProductInstaller);
         appShell = mock(ApplicationShell);
-        pythonExecFactory = mock(PythonExecutionFactory);
-        interpreterService = mock<IInterpreterService>();
-
-        dependencyService = new DataViewerDependencyService(
-            instance(appShell),
-            instance(installer),
-            instance(pythonExecFactory),
-            instance(interpreterService),
-            false
-        );
+        kernel = instance(mock<IKernel>());
+        dependencyService = new DataViewerDependencyService(instance(appShell), false);
+    });
 
-        when(interpreterService.getActiveInterpreter()).thenResolve(interpreter);
-        when(interpreterService.getActiveInterpreter(anything())).thenResolve(interpreter);
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        (instance(pythonExecService) as any).then = undefined;
+    teardown(() => {
+        sinon.restore();
+    });
+
+    test('What if there are no kernel sessions?', async () => {
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        (pythonExecService as any).then = undefined;
-        when(pythonExecFactory.createActivatedEnvironment(anything())).thenResolve(instance(pythonExecService));
+        (kernel.session as any) = undefined;
+
+        const resultPromise = dependencyService.checkAndInstallMissingDependenciesOnKernel(kernel);
+
+        await assert.isRejected(
+            resultPromise,
+            DataScience.noActiveKernelSession(),
+            'Failed to determine if there was an active kernel session'
+        );
     });
+
     test('All ok, if pandas is installed and version is > 1.20', async () => {
-        when(
-            pythonExecService.exec(deepEqual(['-c', 'import pandas;print(pandas.__version__)']), anything())
-        ).thenResolve({ stdout: '0.30.0' });
-        await dependencyService.checkAndInstallMissingDependencies(interpreter);
+        const version = '3.3.3';
+
+        const stub = sinon.stub(helpers, 'executeSilently');
+        stub.returns(Promise.resolve([{ ename: 'stdout', output_type: 'stream', text: version }]));
```

Comment:
I forgot to spy the calls and read what was sent to them -- but, since I agree with you that the imports shouldn't affect the tests, I am just going to revert using Sinon.stubs for the `executeSilently` function.

Reply:
I pushed a change that now verifies the parameters sent to `executeSilently`.